jetty-server/src/main/java/org/eclipse/jetty/server/Request.java

```diff
@@ -410,12 +416,55 @@ public PushBuilder newPushBuilder()
             id = getRequestedSessionId();
         }
 
+        Map<String,String> cookies = new HashMap<>();
+        Cookie[] existingCookies = getCookies();
+        if (existingCookies != null)
+        {        
+            for (Cookie c: getCookies())
+            {
+                cookies.put(c.getName(), c.getValue());
+            }
+        }
+
+        //Any Set-Cookies that were set on the response must be set as Cookies on the
+        //PushBuilder, unless the max-age of the cookie is <= 0
+        HttpFields responseFields = getResponse().getHttpFields();
+        for (HttpField field : responseFields)
+        {
+            HttpHeader header = field.getHeader();
+            if (header == null)
+                continue;
+            switch (header)
+            {
+                case SET_COOKIE:
```

Comment:
You can easily coalesce the null-check and the switch into a single `if (header == SET_COOKIE)`.
How about `SET_COOKIE2` (I'm ok to not consider it, just raising the issue of whether we want to explicitly not consider it).

Reply:
`Set-Cookie2` is an actual header from a valid spec.
https://tools.ietf.org/html/rfc2965